### ISSUE TITLE
Television service?

### ISSUE THREAD

**Issue by vmarks** [Mixiaoxiao/Arduino-HomeKit-ESP8266#218]
Hi, 

Does this support the television service?

I would like to set inputs, volume up/down/mute and power on/off for a television with this project.

**Comment by hassbian-ABC** [comment]
yes

**Comment by PimV5755** [comment]
Pay attention that you cannot control this accessory from the Home app or using Siri.
Other apps wil work. Apple does not allow uncertified (OpenSource HAP) accessories to audio and video. (Shame)

**Comment by vmarks** [comment]
That is incorrect.

Apple allows ravensystems/HAA ir blasters to control television input and
volume for example in home and the control
Center remote control widget.

**Comment by PimV5755** [comment]
That would be great. I tried to make a "speaker" accessory to control the volume of an amplifier/speaker. The accessory paired fine but in the Home-app there came the message that it is not supported. Do you have made it working?

Here ther is also the same problem mentioned (few years ago) 
https://github.com/homebridge/homebridge/issues/1326

**Comment by vmarks** [comment]
I have an IR blaster running ravensystems/haa that controls inputs and
volume of a samsung TV.

Separately, I have homebridge running an onkyo receiver plugin which
appears to homekit like a TV for inputs, and uses a fan speed or lamp
dimmer (config option) to add a volume slider to homekit. Without that, you
have to open control center, tap on remote, and use the volume buttons to
control the volume of the receiver or TV.

but… it works.

**Comment by PimV5755** [comment]
Yes, through Homebridge it works. However, I prefer the native setup so
that I can use Siri. BTW I made a HomeKit “gateway” to control my Denon AVR



Op vr 7 jul. 2023 om 16:39 schreef Victor Marks ***@***.***>

> I have an IR blaster running ravensystems/haa that controls inputs and
> volume of a samsung TV.
>
> Separately, I have homebridge running an onkyo receiver plugin which
> appears to homekit like a TV for inputs, and uses a fan speed or lamp
> dimmer (config option) to add a volume slider to homekit. Without that,
> you
> have to open control center, tap on remote, and use the volume buttons to
> control the volume of the receiver or TV.
>
> but… it works.
>

**Comment by vmarks** [comment]
I agree, I prefer to not use homebridge. But the point I was making was
that open source HAP works with volume.

I would like to replicate your Denon control. Are you able to share how you
did it? I was thinking of using ravensystem/haa and mapping the TV
configuration to the AVR urls that Denon uses.

**Comment by PimV5755** [comment]
Sure. To make it work with Siri I made it a Lightbulb and limited the
control URL to on/off and volume. I did make the double direction work. So
if you use the remote or the volume knob, and the on/off it does update
HomeKit (as it should)
From that you can add more if you go for your solution. I will post an
archive with th sample when I am home.

Op vr 7 jul. 2023 om 17:02 schreef Victor Marks ***@***.***>

> I agree, I prefer to not use homebridge. But the point I was making was
> that open source HAP works with volume.
>
> I would like to replicate your Denon control. Are you able to share how
> you
> did it? I was thinking of using ravensystem/haa and mapping the TV
> configuration to the AVR urls that Denon uses.
>

**Comment by hassbian-ABC** [comment]
I used IRremoteESP8266 and created a TV service

**Comment by hassbian-ABC** [comment]
![Uploading IMG_0113.png…]()

**Comment by hassbian-ABC** [comment]
![Uploading IMG_0114.png…]()

**Comment by hassbian-ABC** [comment]
![IMG_0113](https://github.com/Mixiaoxiao/Arduino-HomeKit-ESP8266/assets/37896214/6398f6a1-0486-4cee-9bd6-d136c01d3036)

**Comment by hassbian-ABC** [comment]
![IMG_0114](https://github.com/Mixiaoxiao/Arduino-HomeKit-ESP8266/assets/37896214/7425481a-4669-4ce1-9d5b-f217c05696a0)

**Comment by PimV5755** [comment]
Nice.

Op vr 7 jul. 2023 om 17:12 schreef img ***@***.***>

**Comment by bartosz-777** [comment]
@hassbian-ABC could you share the code for tv remote(im interested in HomeKit part)? im trying to make it and if you already done it it'd be easier for me
